great_tables/_formats.py

```diff
@@ -2215,26 +2222,36 @@ def fmt_datetime_context(
     if is_na(data._tbl_data, x):
         return x
 
-    # From the date and time format strings, create a datetime format string
-    datetime_format_str = f"{date_format_str}'{sep}'{time_format_str}"
+    if format_str is not None:
+        # If `x` is a string, assume it is an ISO datetime string and convert
+        # it to a datetime object
+        if isinstance(x, str):
```

Comment:
Can you pull this above the block, so it applies to the if and else case? (and then remove the redundant version on 2239 in the else)